detection_rules/rule_validators.py

```diff
@@ -40,12 +40,20 @@ def validate(self, data: QueryRuleData, meta: RuleMeta) -> None:
             packages_manifest = load_integrations_manifests()
             package_integrations = TOMLRuleContents.get_packaged_integrations(data, meta, packages_manifest)
 
+            validation_checks = {"stack": None, "integrations": None}
```

Comment:
👍 so the overall concept is to store return values of both validators and check both at the end for errors.

Reply:
Yes, that is the core change. Instead of calling `raise` on the errors, they are yielded or returned so that `validate()` can make the determination for whether or not to `raise`.